PR: https://github.com/odoo/odoo/pull/187357

From: 0b8bd891fad1c84607fe147a32d705917341fed7
From: Maximilien (malb)
Date: 2024-12-20 09:44:22

Breaking data model changes scores: del:3 + add:0, change matches:
-    name = fields.Char(string='Title', required=True, translate=True)
-    shortcut = fields.Char(string='Abbreviation', translate=True)
-    title: ResPartnerTitle = fields.Many2one('res.partner.title')

Total Changes: 24

[IMP] *: Partner form improvement

This commit will do a lot of change on the partner form:
* Removing res.partner.title model and field
* Removing/modifying some help on fields
* Making the bank_ids list view into an editable tag that has 2 color depending on
if the bank is verified or not
* Add new widget many2many_tags_banks for the banks tags
* Remove some groups and modify existing one
* Change in some placement of smart button on top of the form
* Change in the child_ids of the partner form (naming of the horizontal radio
button, remove input)
* Change the image to fit with v18 ux change
* Warning potential duplicate change
* Hide the country in the cart of the address in the partner form

closes odoo/odoo#187357

Task: 4306429
Related: odoo/enterprise#73879
Related: odoo/upgrade#6779
Related: odoo/upgrade-util#177
Signed-off-by: Florian Gilbert (flg) <flg@odoo.com>

================================= pseudo patch: =================================

--- a/odoo/addons/base/models/res_company.py
+++ b/odoo/addons/base/models/res_company.py
@@ -79,6 +79,7 @@ class ResCompany(models.Model):
     layout_background = fields.Selection([('Blank', 'Blank'), ('Demo logo', 'Demo logo'), ('Custom', 'Custom')], default="Blank", required=True)
     layout_background_image = fields.Binary("Background Image")
     uninstalled_l10n_module_ids = fields.Many2many('ir.module.module', compute='_compute_uninstalled_l10n_module_ids')
+
     _name_uniq = models.Constraint(
         'unique (name)',
         "The company name must be unique!",

--- a/odoo/addons/base/models/res_country.py
+++ b/odoo/addons/base/models/res_country.py
@@ -231,5 +231,8 @@ class ResCountryState(models.Model):
 
     @api.depends('country_id')
     def _compute_display_name(self):
+        if self.env.context.get('hide_country_from_state'):
+            return super()._compute_display_name()
+
         for record in self:
             record.display_name = f"{record.name} ({record.country_id.code})"

--- a/odoo/addons/base/models/res_partner.py
+++ b/odoo/addons/base/models/res_partner.py
@@ -180,15 +180,6 @@ class ResPartnerCategory(models.Model):
         return domain
 
 
-class ResPartnerTitle(models.Model):
-    _name = 'res.partner.title'
-    _order = 'name'
-    _description = 'Partner Title'
-
-    name = fields.Char(string='Title', required=True, translate=True)
-    shortcut = fields.Char(string='Abbreviation', translate=True)
-
-
 class ResPartner(models.Model):
     _name = 'res.partner'
     _description = 'Contact'
@@ -223,7 +214,6 @@ class ResPartner(models.Model):
 
     name = fields.Char(index=True, default_export_compatible=True)
     complete_name = fields.Char(compute='_compute_complete_name', store=True, index=True)
-    title: ResPartnerTitle = fields.Many2one('res.partner.title')
     parent_id: ResPartner = fields.Many2one('res.partner', string='Related Company', index=True)
     parent_name = fields.Char(related='parent_id.name', readonly=True, string='Parent name')
     child_ids: ResPartner = fields.One2many('res.partner', 'parent_id', string='Contact', domain=[('active', '=', True)], context={'active_test': False})
@@ -261,9 +251,9 @@ class ResPartner(models.Model):
     function = fields.Char(string='Job Position')
     type = fields.Selection(
         [('contact', 'Contact'),
-         ('invoice', 'Invoice Address'),
-         ('delivery', 'Delivery Address'),
-         ('other', 'Other Address'),
+         ('invoice', 'Invoice'),
+         ('delivery', 'Delivery'),
+         ('other', 'Other'),
         ], string='Address Type',
         default='contact')
     # address fields
@@ -358,7 +348,9 @@ class ResPartner(models.Model):
         if self.type == 'delivery':
             return "base/static/img/truck.png"
         if self.type == 'invoice':
-            return "base/static/img/money.png"
+            return "base/static/img/bill.png"
+        if self.type == 'other':
+            return "base/static/img/puzzle.png"
         return super()._avatar_get_placeholder_path()
 
     def _get_complete_name(self):
